### DISCUSSION TITLE
Custom label and custom value

### DISCUSSION THREAD

**Discussion by ZareefJafar** [mjfryc/mjaron-tinyloki-java#23] [Q&A] [answered]
Hi @azplanlos @mjfryc ,
I am building a Java Spring Boot application where I implemented tinyloki and loved it. I want to use the same label but different values for different `Controller Classes`. But I do not want to create `LogController logController` and ` ILogStream stream = logController.createStream` repeatatively in different classes. How to implement this?

Thanks

**Comment by mjfryc** [comment, marked as answer]
Hi @ZareefJafar ,
Very, very sorry for lack of my response. I missed this message, expecting that GitHub will notify me but I didn't receive any notification.

Hope that you found any matching solution for you.

* Currently this _tinyloki_ allows creating streams with static (fixed) labels only, and it is a definition of the stream (value of label is not changed),
* but Loki provides sth named _structured metadata_ and I will investigate it in the future. I've created task for it: #26.
* Note that structured metadata may be not enabled in Grafana Loki configuration and may not work: <https://grafana.com/docs/loki/latest/get-started/labels/structured-metadata/#enable-or-disable-structured-metadata>

Regards,
MJ

**Comment by mjfryc** [comment]
Hi @ZareefJafar,
So, finally... structured metadata is supported now :D
Issue: #26 
Regards,
MJ
